geomstats/numerics/geodesic_solvers.py

```diff
@@ -599,7 +599,7 @@ def __call__(self, x):
         x_t = gs.stack(
             [gs.power(x, power) for power in range(self.degree + 1)], axis=-1
         )
-        return gs.einsum("...j,nj->n...", self.coeffs, x_t)
+        return gs.einsum("...kj,nj->...nk", self.coeffs, x_t)
```

Comment:
I think this one was right, given our purpose:

* `coeffs` can be 1d or 2d: it basically tells us `dim`
* `x` is always 1d

The goal is to solve for multiple dimensions simultaneously for the same `t`.

Am I missing anything?

Reply:
The old implementation won't work for cases with multiple points (`coeffs` is 3d), so when `poly(t)` is called in `geodesic_bvp`, the output shape is not correct (normally we want shape `(n_points, time, dim)`). This will make it work for multiple points and it doesn't require any other changes for other parts.

Reply:
btw, the code in `_solve`, `_objective` and so on assumes we are solving only for one point at a time. we can actually solve for all simultaneously. I may create that version before we merge.

Reply:
I agree to what you said here, I think the problem is that in `_update_poly_coeffs`, we have a `reshape(dim,degree-1)` so `coeffs` is always somewhat 2d (whether `(1,deg+1)` or `(n,deg+1)`) for one point.

As you've said, we were solving one `log` at a time but the problem I had was that we were solving `geodesic_bvp` for multiple points at a time, in this case `coeffs.shape=(n_points,dim,deg)` and `x_t.shape=(n_times,deg)` and so the old subscripts produces `output.shape=(n_times,n_points,dim)` instead of `output.shape=(n_points,n_times,dim)`.

I think it may be better to unify the use of `_Polynomial` by also solving `log` for multiple points simutaneously.

Reply:
I also think we should not have `coeffs` or output of `poly` being 1d because points of one dimensional distribution now cannot be represented by a float but by a `(1,)` array. This change will make things more coherent.

Reply:
Thanks @tramy1258. I've done changes reflecting your suggestions!